### PR TITLE
restore support for accessors & mutators

### LINKED ISSUE #353: Accessor methods not used for translations
Translated attributes doesn't run accessor methods. 
If have `public $translatable = ['title']`, the method `getTitleAttribute` isn't accessible from `$model->title`.

Is there another way to get accessors with translated attributes ?

Thanks

### PR DESCRIPTION
**Pull request by mjauvin**
since the hook to the "model.beforeGetAttribute" event is returning a value, the model accessors/mutators were not triggered anymore.

fixes #353

### CODE REVIEW

**Comment by mjauvin** [comment]
I guess we should do the same for mutators...

**Comment by mjauvin** [comment]
I noticed there is no `mutateAttribute()` method in `October\Rain\Database\Model` like there is in `Halcyon\Model`...

instead of the ugly `$model->{'get'.Str::studly($key).'Attribute'}($value)` call, we could use `$model->mutateAttribute($key, $value)` like in Halcyon\Model

**Comment by mjauvin** [comment]
@alvaro-canepa could you test this PR to see if it solves your problem?

File: classes/TranslatableBehavior.php [hunk 1]
```diff
@@ -58,15 +59,25 @@ public function __construct($model)
 
         $this->initTranslatableContext();
 
-        $this->model->bindEvent('model.beforeGetAttribute', function($key) {
-            if ($key !== 'translatable' && $this->isTranslatable($key)) {
-                return $this->getAttributeTranslated($key);
+        $this->model->bindEvent('model.beforeGetAttribute', function($key) use ($model) {
```
Comment (bennothommo) [inline classes/TranslatableBehavior.php]: @mjauvin Space between `function` and `(` please :)
```suggestion
        $this->model->bindEvent('model.beforeGetAttribute', function ($key) use ($model) {
```

File: classes/TranslatableBehavior.php [hunk 2]
```diff
@@ -58,15 +59,25 @@ public function __construct($model)
...
+            if ($this->isTranslatable($key)) {
+                $value = $this->getAttributeTranslated($key);
+                if ($model->hasGetMutator($key)) {
+                    $method = 'get'.Str::studly($key).'Attribute';
```
Comment (bennothommo) [inline classes/TranslatableBehavior.php]: @mjauvin Add spaces between concatenation parts.
```suggestion
                    $method = 'get' . Str::studly($key) . 'Attribute';
```

File: classes/TranslatableBehavior.php [hunk 3]
```diff
@@ -58,15 +59,25 @@ public function __construct($model)
...
+                    $value = $model->{$method}($value);
+                }
+                return $value;
             }
         });
 
-        $this->model->bindEvent('model.beforeSetAttribute', function($key, $value) {
-            if ($key !== 'translatable' && $this->isTranslatable($key)) {
-                return $this->setAttributeTranslated($key, $value);
+        $this->model->bindEvent('model.beforeSetAttribute', function($key, $value) use ($model) {
```
Comment (bennothommo) [inline classes/TranslatableBehavior.php]: @mjauvin Space between `function` and `(`.
```suggestion
        $this->model->bindEvent('model.beforeSetAttribute', function ($key, $value) use ($model) {
```

File: classes/TranslatableBehavior.php [hunk 4]
```diff
@@ -58,15 +59,25 @@ public function __construct($model)
...
+            if ($this->isTranslatable($key)) {
+                $value = $this->setAttributeTranslated($key, $value);
+                if ($model->hasSetMutator($key)) {
+                    $method = 'set'.Str::studly($key).'Attribute';
```
Comment (bennothommo) [inline classes/TranslatableBehavior.php]: @mjauvin Spaces between concatenation parts.
```suggestion
                    $method = 'set' . Str::studly($key) . 'Attribute';
```

**Comment by alvaro-canepa** [comment]
> @alvaro-canepa could you test this PR to see if it solves your problem?

Great! I will check this!

**Comment by bennothommo** [comment]
@mjauvin Looks great. Just needs someone who uses the Translate plugin to test it out, and then it's good to go.

**Comment by mjauvin** [comment]
@alvaro-canepa did you get a chance to test this yet?

**Comment by alvaro-canepa** [comment]
> @alvaro-canepa did you get a chance to test this yet?

Sorry, I still haven't been able to test it. :(

**Comment by mjauvin** [comment]
@alvaro-canepa we'd like to merge and close this PR to release v1.6.3. Do you think you can take 5m to test it?

Thanks.

**Pull request merged**